### PR TITLE
Refactored ProjectionUtilities.h.

### PR DESCRIPTION
**Pull request by odlomax**
Refactored ProjectionUtilities.h to use eckit coordinate transforms.

Moved mirror_latlon to CubedSphereEquiDistFV3Projection.cc.

Addressed other minor comments from Willem.

All tests pass in release build.

### CODE REVIEW

File: src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc [hunk 1]
```diff
@@ -27,6 +27,58 @@ namespace atlas {
 namespace projection {
 namespace detail {
 
+// Helper functions.
+namespace {
+
+// -------------------------------------------------------------------------------------------------
+
+static void mirror_latlon(const double lonlat2[], const double lonlat3[], const double lonlat1[],
```
Comment (MarekWlasak) [inline src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc]: Does this now  need to be static?

**Comment by MarekWlasak** [comment]
@odlomax - how did you check this?

File: src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc (hunk 1 above)
Comment (MarekWlasak) [reply src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc]: I think it can be removed.

**Review by danholdaway — APPROVED** [review]

File: src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc (hunk 1 above)
Comment (odlomax) [reply src/atlas/projection/detail/CubedSphereEquiDistFV3Projection.cc]: I think you're right. :-)

**Review by MarekWlasak — APPROVED** [review]

**Comment by odlomax** [comment]
> @odlomax - how did you check this?

Made sure all the tests passed and eyeballed the gmsh files. 
cs_xyz

**Pull request merged**